pkg/controllers/csi/provisioner/controller.go

```diff
@@ -92,187 +80,58 @@ func (provisioner *OneAgentProvisioner) SetupWithManager(mgr ctrl.Manager) error
 func (provisioner *OneAgentProvisioner) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	log.Info("reconciling DynaKube", "namespace", request.Namespace, "dynakube", request.Name)
 
-	dk, err := provisioner.getDynaKube(ctx, request.NamespacedName)
+	var dk dynakube.DynaKube
+
+	err := provisioner.apiReader.Get(ctx, request.NamespacedName, &dk)
 	if err != nil {
 		if k8serrors.IsNotFound(err) {
-			return reconcile.Result{}, provisioner.db.DeleteDynakube(ctx, request.Name)
+			return reconcile.Result{}, nil // TODO: Call GC here
 		}
 
 		return reconcile.Result{}, err
 	}
 
```

Comment:
We no longer check if the download of the package is required. Even in appliation monitoring mode without csi driver we download the binaries.

Reply:
this is not really a problem, as in `main` we use the CSI-driver if it is there
the feature-branch is not up to date in this respect

Reply:
Don't we still have the `useCSIDriver` field and want to support it, even though the CSI driver is installed? 🤔

Fine for me then

Reply:
not in main (already in 1.4)

The "rebase" will be fuuuun 😅